src/System.Net.Primitives/src/System/Net/CredentialCache.cs

```diff
@@ -167,6 +161,11 @@ public NetworkCredential GetCredential(Uri uriPrefix, string authenticationType)
                 GlobalLog.Print("CredentialCache::GetCredential(uriPrefix=\"" + uriPrefix + "\", authType=\"" + authenticationType + "\")");
             }
 
+            if (_cache == null)
+            {
+                return null;
+            }
+
```

Comment:
>This ends up changing the logging by short-circuiting the logging call at the end of the function.

I recommend that logging calls be added to these places that 'short-circuit' the original code paths.